src/rvsdg/egglog_optimizer/conditional_invariant_code_motion.egg

```diff
@@ -0,0 +1,50 @@
+; If (= x (prefix-containing-operand (Gamma ...) e)), then the first x outputs
+; of Gamma contain e.
+(function prefix-containing-Operand (Body Operand) i64 :merge (max old new))
+(rule
+    ((= gamma (Gamma pred inputs outputs))
+     (Body-contains-Operand gamma 0 e))
+    ((set (prefix-containing-Operand gamma e) 0))
+    :ruleset fast-analyses)
+(rule
+    ((= gamma (Gamma pred inputs outputs))
+     (= i (prefix-containing-Operand gamma e))
+     (Body-contains-Operand gamma i e))
+    ((set (prefix-containing-Operand gamma e) (+ i 1)))
+    :ruleset fast-analyses)
+
+; Limit blowup by only repeatedly hoisting so many times and by limiting the
+; number of branches we hoist from or max number of inputs we can add.
+(let max-hoists 1)
+(function hoist-count (Body) i64 :merge (max old new))
+(rule ((= gamma (Gamma pred inputs outputs)))
+      ((set (hoist-count gamma) 0))
+      :ruleset fast-analyses)
+(let max-branches 2)
+(let max-inputs 4)
+
+; Hoist code out of a gamma, adding it as an argument instead.
+(rule 
+    ((= gamma (Gamma pred (VO inputs) outputs))
+     ; All branches contain hoist-me
+     (= (prefix-containing-Operand gamma hoist-me)
+        (VecVecOperand-length outputs))
+     (!= hoist-me (Arg any))
```

Comment:
The semantics here is not what you intended. Here, if there exists an `any` such that `(!= host-me (Arg any))`, this check will pass. @clyben is working on something like a `not-an-arg` analysis that will be helpful

Reply:
did this get resolved?

Reply:
No, although this isn't a soundnes problem, it will just cause some useless single-arg hoists. Once @clyben merges `not-an-arg` I'll be sure to update.